### PR TITLE
Adjust GitHub Actions config to avoid double testing on PRs

### PR DESCRIPTION
**Pull request by blag**
Our current GitHub Actions configuration runs tests twice for each push - once when the branch is pushed, and once more when a pull request is opened, reopened, or edited.

This is inefficient and slows down development since people have to wait for more tests to finish.

This PR slightly tweaks the GHA config to only run tests on pushes to:

* `master` - ensure post-merge code is good
* `v*` branches - version branches that track minor versions (eg: `v3.4`)
* `v*` tags - version tags that track bugfix releases (eg: `v3.4.0`), typically are tagged on commits to version branches

I have included comments on the configuration to ensure that everybody who isn't familiar with GHA understands what the intent is.

### CODE REVIEW

**Review by cognifloyd — APPROVED** [review]
LGTM

File: .github/workflows/ci.yaml
Comment (Kami) [inline .github/workflows/ci.yaml:14]: Technically we could / should probably also filter pull requests so builds are only triggered if it's a PR against master branch (and perhaps a version one?).

File: st2common/st2common/models/db/execution.py
Comment (Kami) [inline st2common/st2common/models/db/execution.py:101]: Please add a comment on why we need this pylint pragma and hopefully we can figure out root cause in the near future.

**Comment by Kami** [comment]
I added branches filter to PR trigger and I will merge it once the build passes so I can sync and merge rest of my PRs.

File: st2common/st2common/models/db/execution.py
Comment (blag) [reply st2common/st2common/models/db/execution.py:101]: I don't know the root cause. pylint just randomly decided to not like that line, so I dug through the code, and checked the mongoengine documentation to make sure that the object would have an `id` attribute by the time that code is evaluated.

**Comment by amanda11** [comment]
LGTM

**Review by amanda11 — APPROVED** [review]

**Review by Kami — APPROVED** [review]

**Comment by armab** [comment]
closes https://github.com/StackStorm/st2/issues/5144

**Pull request merged**
